tests/utils.go

```diff
@@ -0,0 +1,17 @@
+package tests
+
+import (
+	"fmt"
+
+	. "github.com/onsi/gomega"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	
+	"knative.dev/pkg/apis"
+)
+
+func ExpectPipelineRunNotToFail(pr *v1beta1.PipelineRun) {
+	failed := pr.IsDone() && pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsFalse()
+	if failed {
+		Expect(failed, fmt.Sprintf("did not expect pr %s:%s to fail", pr.Namespace, pr.Name)).NotTo(BeTrue())
+	}
+}
```

Comment:
yep I also think that we should probably move this function somewhere else. I'd suggest following:
1. move this function to the existing [utils pkg](https://github.com/redhat-appstudio/e2e-tests/blob/main/pkg/utils/util.go)
2. update the function in a following way
```
func PipelineRunSucceeds(pr *v1beta1.PipelineRun) (bool, error) {
	if pr.IsDone() && pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsTrue() {
		return true, nil
	}
	return false, fmt.Errorf("did not expect PipelineRun %s:%s to fail", pr.GetNamespace(), pr.GetName())
}
```
3. Then from the test we could call it like
```
Expect(PipelineRunSucceeds(pr)).To(BeTrue())
```
According to [docs of the Expect function](https://pkg.go.dev/github.com/onsi/gomega#Expect) - it passes the first argument to the matcher and the rest of the arguments are expected to be nil/zero.
So in case there will be an error returned from `PipelineRunSucceeds()`, we will get the error on the test output.

I think this might be the way we could update other helper functions or [functions passed to `Eventually`](https://github.com/redhat-appstudio/e2e-tests/blob/377907087ec7a4436933f83112e0594f13328567/tests/build/build.go#L131-L142) that we are passing into a matcher with only a boolean value - which then produces an error, but not very helpful for debugging.

Wdyt?

Reply:
or you know what, let's not hold this PR anymore only due to my silly request and I'll create a follow-up PR and ping you guys for a review.